### ISSUE TITLE
"incorrect format" error on Linux

### ISSUE THREAD

**Issue by josvernon** [dotnet/core#2831]
I'm running ASP.NET Blazor code referencing .NET Core 3.0.0-preview5-27626-15.

The base project works fine on both Windows and also Azure Linux

One of my references is to System.Configuration.ConfigurationManager. 

    <Reference Include="System.Configuration.ConfigurationManager">
      <HintPath>..\..\..\..\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App\3.0.0-preview5-27626-15\System.Configuration.ConfigurationManager.dll</HintPath>
    </Reference>

As soon as I put code in which uses something in this namespace this it blows up on Linux. Takes down the entire web server. It works fine on Windows. 

The Azure Linux Server log says:

2019-06-05T13:26:37.724864005Z An attempt was made to load a program with an incorrect format.
2019-06-05T13:26:37.724872905Z  (0x8007000B)
2019-06-05T13:26:37.724894306Z    at System.Reflection.RuntimeModule.GetTypes(RuntimeModule module)
2019-06-05T13:26:37.724918107Z    at System.Reflection.RuntimeAssembly.get_DefinedTypes()

Full stack trace attached.
[log.txt](https://github.com/dotnet/core/files/3257432/log.txt)

**Comment by wfurt** [comment]
You can try it with:
```
export COREHOST_TRACE=1  
export COMPlus_ZapDisable=1   
```

cc: @ericstj in case this is another weird packaging issue.

**Comment by omajid** [comment]
It looks like you are trying to make use of `..\..\..\..\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App\3.0.0-preview5-27626-15\System.Configuration.ConfigurationManager.dll` which looks a lot like a Windows path? I expect this contains crossgen'ed code, which might not work on non-Windows platforms. Will using a normal nuget package work here?

**Comment by wfurt** [comment]
good point @omajid I did not expect anybody using Windows assemblies on Linux. If this is root cause,  COMPlus_ZapDisable=1  should make difference.

**Comment by josvernon** [comment]
I would imagine that as a cross platform thing that the publishing would work whatever platform I pushed to. But yes thanks I will try the zap thing and let you know what I find. :-)

**Comment by wfurt** [comment]
The "cross platform" means that you can target different OSes but it does not give binary compatibility. Aside from crossgen (when we pre-compile to native some parts), some assemblies use different source code for different platforms.

**Comment by svick** [comment]
@josvernon It should work as long as you do it correctly: by referencing the NuGet package, instead of a file on disk.

**Comment by josvernon** [comment]
In Azure I went to the configuration section, added an environment variable with the name COMPlus_ZapDisable and with a value of 1. Restarted the instance and tried again. No difference to the error.

I also removed the assembly reference and replaced it with a reference to the System.Configuration.ConfigurationManager NuGet package. That fixed the issue.

However that just shifts the problem over a bit. Let me see if I can explain.

So...

Let's take a call to,

var tf = new System.Windows.Media.Typeface("Verdana");

... for which I require a reference to PresentationCore.dll.

If I compile with just the normal "Microsoft.NETCore.Platforms(3.0.0-preview5-27626-15)" and "Microsoft.NETCore.Targets(3.0.0-preview5-27626-15)" NuGet packages I cannot reference System.Windows.Media because these packages don't include PresentationCore.dll.

I can add the NuGet package Microsoft.Xaml to get PresentationCore.DLL. However that package was last updated in 2016 and so I get an error when .NET Core attempts to load up this rather ancient copy.

I know there are .NET Core 3.0 DLLs on my machine at,

"Program Files\dotnet\shared\Microsoft.WindowsDesktop.App\3.0.0-preview5-27626-15\".

If I reference WindowsBase.dll and PresentationCore.dll at those locations it works on Windows. Note that the WindowsBase.dll reference is required otherwise Windows will pull in some random and incompatible 4.2 version rather than the 4.8 version in .NET Core 3.0.

I want to deploy to Azure. Azure doesn't have .NET Core 3.0 on it so I need the publish option deployment mode to be self contained.

However it appears that Visual Studio doesn't understand that some assemblies may be Windows specific and thus is is quite happy to deploy them on Linux even if they don't work.

In short...

How is one to know which references to use? 

I cannot rely on the .NET Core 3.0 assemblies on my machine because they are not reliably cross platform.

I cannot rely on assemblies obtained via NuGet because they are not reliably compatible with .NET Core 3.0.

It's a conundrum. Am I missing something?

**Comment by karelz** [comment]
That maybe a good feedback for Visual Studio - please use "Send Feedback" from inside VS to provide that suggestion.
We do not track VS issues on GitHub. Thanks!

Closing as it has been resolved.
